src/lib/mina_lib/mina_lib.ml

```diff
@@ -1392,6 +1487,51 @@ let send_resource_pool_diff_or_wait ~rl ~diff_score ~max_per_15_seconds diff =
   able_to_send_or_wait ()
 
 let create ?wallets (config : Config.t) =
+  let _precomputed_block_writer_setup =
+    let open Option in
+    (* appending *)
+    iter config.precomputed_blocks_file ~f:(fun file ->
```

Comment:
It appears that this value is only used to initialize `precomputed_block_writer.appending`. Can you restructure this to reduce the complexity here?